File: src/com/colacelli/irclib/connection/listeners/OnRawCodeListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;

public interface OnRawCodeListener extends Listener {
    Type TYPE = Type.RAW_CODE;

    int rawCode();
    void onRawCode(Connection connection, String message, int rawCode, String... args);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/com/colacelli/irclib/connection/listeners/OnRawCodeListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection

interface OnRawCodeListener : Listener {
    fun rawCode(): Int
    fun onRawCode(connection: Connection, message: String, rawCode: Int, args: List<String>)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.RAW_CODE
    }
}
